File: 2012/P1.c
#include <stdio.h>

int main()
{
    int N, i, indexLen, tempInt, a, b, sum;
    
    scanf("%d", &N);
    
    int lst[N+1];
    
    sum = 0;
    for(i=0; i<N; i++)
    {
        scanf("%d", &tempInt);
        lst[i] = tempInt;
        sum = sum + tempInt;
    }
    lst[i+1] = 0;
    
    a = sum - lst[0];
    b = 0;
    i = 0;
    while(i<N)
    {
        if(a==b)
        {
            printf("%d", i);
        }
        b = b + lst[i];
        a = a - lst[i+1];
        i ++;
    }
    printf("\n");
}

File: 2012/P1.py
# 2012 Problem 1
# Equilibrium Index

N = int(input())

lst = [int(i) for i in input().split()]
lst.append(0)
a = sum(lst)-lst[0]
b = 0


indexes = []

i = 0
while i < N:
    
    if a == b:
        indexes.append(i)
    b += lst[i]
    a -= lst[i+1]
    i+=1

indexLen = len(indexes)

for i in range(indexLen):
    print(indexes[i], end='')
    if i!= indexLen-1:
        print(" ", end='')
print()
